### PR TITLE
[FRONT-140] Added pagination to overview page

### PR DESCRIPTION
**Pull request by mohammad-akkad**
## Description of Issue
add pagination in TrendingProjectsQuery and implement it in the overview and compare pages
## How I implemented
Changed the query and ran codegen, then I added previous and next buttons to the pages

### CODE REVIEW

**Review by ArnoClaude — CHANGES_REQUESTED** [review]
@mohammad-akkad Could you make sure the Vercel Deployment doesn't fail? I'll review later.

**Review by kigadye — CHANGES_REQUESTED** [review]
Very nice! Couple things could be improved though:

- [ ] Set the page size/ number of entries per request back to 30 (pagination should only apply for pages that have more than 30 results)
- [ ] Change the appearance of the prev/ next buttons if they're disabled (just like details page nav)
- [ ] Either find a way to display current + total number of entries/ pages in filterbar, or remove "showing 5/5 entries"

**Review by kigadye — CHANGES_REQUESTED** [review]
- [x] Don't show next/ prev buttons if there is only 1 page
- [x] Remove all added eslint ignore rules and fix the underlying issues if any
- [ ] Total count sometimes doesn't update when you change filters (probably due to caching). Also, I saw you're just using the length of the query result without pagination, but that would always return 30 at most, which is the issue we had previously. Idk how to fix this off the top of my head and it's not prio 1 for the demo so I would say just leave the total count as it was before, i.e. use the length of the filtered query with pagination although it's wrong and add a `@TODO` comment to fix it later.

File: src/components/page/details/RightSidebar.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ const RightSidebar = ({ project }: Props) => {
         <Sidebar.Box.GithubStatItem Icon={FaGithub} link={project.githubUrl as string} />
       </Sidebar.Box>
 
-      {project.languages?.length > 0 && (
+      {project.languages?.length && project.languages?.length > 0 && (
```
Comment (ArnoClaude) [inline src/components/page/details/RightSidebar.tsx:60]: `project.languages?.length` is enough

File: src/components/page/overview/FilterBar.tsx [hunk 2]
```diff
@@ -10,45 +13,85 @@ type FilterBarProps = {
...
-    <div className="flex flex-row gap-3">
-      {/* Sorting */}
-      {sorting && <SortModal sorting={sorting} setSorting={setSorting} />}
+  updateFilters,
+  pageInfo,
+  setPagination
+}: FilterBarProps) => {
+  const handleClickLeft = () => {
+    setPagination({
+      last: 30,
```
Comment (ArnoClaude) [inline src/components/page/overview/FilterBar.tsx:31]: magic number, could be `PAGE_SIZE`

File: src/components/page/overview/FilterBar.tsx [hunk 3]
```diff
@@ -10,45 +13,85 @@ type FilterBarProps = {
...
-        .map((key) => (
-          <FilterItemModal
-            key={key}
-            currentKey={key as FilterOption['key']}
-            filters={filters}
-            updateFilters={updateFilters}
-          />
-        ))}
-    </div>
+  return (
```
Comment (ArnoClaude) [inline src/components/page/overview/FilterBar.tsx:47]: Currently, all FilterItemModal are being created at each render. Consider using useMemo to prevent unnecessary rerendering if the filters and updateFilters props haven't changed.

```
const filterItemModals = useMemo(() => {
  return Object.keys(filters)
    .filter((key) => !Object.values(TimeFilterOption).includes(key as TimeFilterOption))
    .map((key) => (
      <FilterItemModal
        key={key}
        currentKey={key as FilterOption['key']}
        filters={filters}
        updateFilters={updateFilters}
      />
    ))
}, [filters, updateFilters]);

// Then, in your render...
<div className="flex flex-row gap-3">
  {sorting && <SortModal sorting={sorting} setSorting={setSorting} />}
  {sorting && Object.keys(filters).length > 0 && <div className="my-auto h-4/5 border-l border-gray-800" />}
  {filterItemModals}
</div>

```

**Review by ArnoClaude — CHANGES_REQUESTED** [review]
Good stuff @mohammad-akkad . Left you three small improvement suggestions :)

File: src/pages/bookmarks.tsx [hunk 4]
```diff
@@ -36,6 +45,20 @@ const Bookmarks = () => {
     (edge) => edge.node.project?.id as string
   ) as string[]
 
+  // Fetch data from Supabase using generated Urql hook for total count
+  const [{ data: urqlDataTotal }] = useTrendingProjectsQuery({
```
Comment (ArnoClaude) [inline src/pages/bookmarks.tsx:49]: Perhaps I'm missing something, but is this second call to `useTrendingProjectsQuery` necessary? Can't you get the total amount of rows from the other `useTrendingProjectsQuery` call?

**Review by ArnoClaude — CHANGES_REQUESTED** [review]
Awesome job. Left one remark :)

File: src/pages/bookmarks.tsx (hunk 4 above)
Comment (mohammad-akkad) [reply src/pages/bookmarks.tsx:49]: good catch, yeah I removed the pagination params as they affect the total count, otherwise we can yeah as you said use the same one

File: src/pages/index.tsx [hunk 5]
```diff
@@ -2,49 +2,77 @@ import { useState, useEffect } from 'react'
...
+  })
+
+  // Fetch data from Supabase using generated Urql hook for total count
+  const [{ data: urqlDataTotal }] = useTrendingProjectsQuery({
+    variables: {
+      orderBy: sorting || defaultSort,
+      filter: {
+        ...filters
+      },
+      ...pagination
```
Comment (kigadye) [inline src/pages/index.tsx:49]: I believe this doesn't make sense because Supabase returns max 30 entries in any case, and it's just the same query as the one before, so you're doubling the data you fetch but don't get any additional information. Not sure if/ how it's possible to get the total count from Supabse, so let's just remove the unnecessary query and use the length of the current page for now

File: src/components/side-effects/Details/index.tsx [hunk 6]
```diff
@@ -163,7 +163,7 @@ const Details = ({ id }: DetailsProps) => {
             } / ${project.name as string}`}
             url={project.githubUrl as string}
             explanation={project.eli5 || 'No explanation'}
-            about={project.about || 'No description'}
+            about={(project.about as string) || 'No description'}
```
Comment (kigadye) [inline src/components/side-effects/Details/index.tsx:166]: no need to cast to string here because we have a fallback - if about is not present we want to show "no description", not an empty string

File: src/components/side-effects/Compare/index.tsx [hunk 7]
```diff
@@ -45,6 +47,13 @@ const Compare = ({ category }: CompareProps) => {
   const [columnVisibility, setColumnVisibility] = useState({})
   const [slackLoading, setSlackLoading] = useState(false)
   const [notificationStatus, setNotificationStatus] = useState<'success' | 'error' | ''>('')
+  const [pageInfo, setPageInfo] = useState<PageInfo>()
+  const [pagination, setPagination] = useState<paginationParameters>({
+    first: 10,
```
Comment (kigadye) [inline src/components/side-effects/Compare/index.tsx:52]: should probably use `PAGE_SIZE` instead of `10` here

**Review by kigadye — CHANGES_REQUESTED** [review]
When I run the project locally I get the following error:
Screenshot 2023-06-28 at 01 38 35

**Review by kigadye — APPROVED** [review]

**Review by pczern — APPROVED** [review]
needed for handover tomorrow
